spec/integration/integration_spec.rb

```diff
@@ -133,12 +133,29 @@ def stop_server
       @reply_to = nil
       client.on :pong do |data|
         @reply_to = data.reply_to
+        queue.push nil
         client.stop!
       end
       start_server
-      wait_for_server
+      queue.pop_with_timeout(5)
       expect(@reply_to).to be 1
     end
+    it 'rebuilds the websocket connection when dropped' do
```

Comment:
I'm not super happy with this test, but it's been the most consistent way I've been able to validate that the reconnect works as expected for each of the 3 supported concurrency libraries.